src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets

```diff
@@ -292,5 +296,26 @@
   
   <!-- This target collects all the resolved references that are used to actually compile. -->
   <Target Name="CollectResolvedCompilationReferencesDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(ReferencePathWithRefAssemblies)" />
+  
+  <!-- This target collects all the things built by the project for the up to date check. -->
+  <!-- See CopyFileToOutputDirectory target -->
+  <Target Name="CollectBuiltDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(UpToDateCheckBuilt)">
+    <ItemGroup>
+      <!-- Assembly output, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(CopyBuildOutputToOutputDirectory)' != 'false' and '$(SkipCopyBuildProduct)' != 'true'" Include="$(TargetPath)"/>
+      <UpToDateCheckBuilt Include="@(IntermediateAssembly)"/>
+
+      <!-- Documentation file, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(_DocumentationFileProduced)'=='true'" Include="@(FinalDocFile)"/>
+      <UpToDateCheckBuilt Condition="'$(_DocumentationFileProduced)'=='true'" Include="@(DocFileItem)"/>
+
+      <!-- Symbols, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(_DebugSymbolsProduced)'=='true'" Include="@(_DebugSymbolsIntermediatePath)"/>
+      <UpToDateCheckBuilt Condition="'$(_DebugSymbolsProduced)'=='true' and '$(SkipCopyingSymbolsToOutputDirectory)' != 'true' and '$(CopyOutputSymbolsToOutputDirectory)' != 'false'" Include="@(_DebugSymbolsOutputPath)"/>
+      
+      <!-- app.config -->
+      <UpToDateCheckBuilt Condition=" '@(AppConfigWithTargetPath)' != '' " Include="@(AppConfigWithTargetPath->'$(OutDir)%(TargetPath)')" Original="@(AppConfigWithTargetPath)"/>
+    </ItemGroup>
```

Comment:
Coincidentally, I'm having an issue right now where my changes to Content with CopyToOutputDirectory=PreserveNewest don't affect up-to-date-ness. For dotnet/sdk, these are the .targets files and it's very common to change them without changing code. I currently have to make a dummy edit to a .cs file in order for tests to pick up my target changes.

Questions:
1. Does this fix that? I'm not seeing it.
2. If not, is there a bug already on it?

Reply:
If this is always how it worked in the old .csproj system, then I think it would be OK to say "by design" here. I could add UpToDateCheckBuilt items in my project once this goes in as the workaround.

Reply:
No, that sounds like a bug. File an issue and I'll take a look.

---
In reply to: [161854343](https://github.com/dotnet/project-system/pull/3142#discussion_r161854343) [](ancestors = 161854343)

Reply:
Nevermind. Our csproj has a manual `Copy` task invocation in a target instead of using `Content`. I remembered incorrectly how it was implemented. I did find a completely different bug while working on a minimal repro: https://github.com/dotnet/project-system/issues/3152